WeScan/Scan/ScannerViewController.swift

```diff
@@ -137,46 +143,94 @@ final class ScannerViewController: UIViewController {
+            let window = UIApplication.shared.keyWindow
+            
+            toolbarConstraints = [
+                toolbar.widthAnchor.constraint(equalTo: view.widthAnchor),
+                toolbar.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+                toolbar.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+                toolbar.topAnchor.constraint(equalTo: view.topAnchor)
+            ]
+            
+            if let safeAreaInsets = window?.safeAreaInsets {
+                toolbarConstraints.append(toolbar.heightAnchor.constraint(equalToConstant: safeAreaInsets.top + 44.0))
+            }
+            
+            cancelButtonConstraints = [
+                cancelButton.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 24.0),
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: cancelButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+            ]
+            
+            let shutterButtonBottomConstraint = view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: shutterButton.bottomAnchor, constant: 8.0)
+            shutterButtonConstraints.append(shutterButtonBottomConstraint)
+        } else {
+            toolbarConstraints = [
+                toolbar.widthAnchor.constraint(equalTo: view.widthAnchor),
+                toolbar.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                toolbar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                toolbar.topAnchor.constraint(equalTo: view.topAnchor)
+            ]
+            
+            cancelButtonConstraints = [
+                cancelButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 24.0),
+                view.bottomAnchor.constraint(equalTo: cancelButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+            ]
+            
+            let shutterButtonBottomConstraint = view.bottomAnchor.constraint(equalTo: shutterButton.bottomAnchor, constant: 8.0)
+            shutterButtonConstraints.append(shutterButtonBottomConstraint)
+        }
+        
+        NSLayoutConstraint.activate(quadViewConstraints + cancelButtonConstraints + shutterButtonConstraints + activityIndicatorConstraints + toolbarConstraints)
     }
     
     // MARK: - Actions
     
+    @objc private func orientationChanged(_ notification: NSNotification) {
+        var rotation: CGFloat = 0
+        switch UIDevice.current.orientation {
+        case .portrait, .faceUp:
+            rotation = 0
+        case .landscapeLeft:
+            rotation = .pi / 2
+        case .landscapeRight:
+            rotation = -(.pi / 2)
+        case .portraitUpsideDown, .faceDown:
+            rotation = .pi
+        default:
+            break
+        }
+        
+        let transform = CGAffineTransform(rotationAngle: rotation)
+        UIView.animate(withDuration: 0.4, delay: 0.0, options: .beginFromCurrentState, animations: {
```

Comment:
The button should also be translated a bit. Right now it rotates and then is weirdly placed: too far away from the edge of the screen.

Reply:
Hey, I'm actually gonna remove the Cancel button transform, as we can't transform the `UIBarButtonItem`s and I feel it'll be a pain to maintain in the future. We can always make a new PR later on 🙂